Lib/test/test_ssl.py

```diff
@@ -2238,6 +2246,53 @@ def test_transport_eof(self):
             self.assertRaises(ssl.SSLEOFError, sslobj.read)
 
 
+@requires_tls_version('TLSv1_3')
+class SimpleBackgroundTestsTLS_1_3(unittest.TestCase):
+    """Tests that connect to a simple server running in the background."""
+
+    def setUp(self):
+        ciphers = [cipher['name'] for cipher in ctx.get_ciphers()
+                   if cipher['protocol'] == 'TLSv1.3']
+
+        self.matching_cipher = ciphers[0]
+        self.mismatched_cipher = ciphers[-1]
+
+        self.server_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+        self.server_context.set_ciphersuites(self.matching_cipher)
+        self.server_context.load_cert_chain(SIGNED_CERTFILE)
+        server = ThreadedEchoServer(context=self.server_context)
+        self.enterContext(server)
+        self.server_addr = (HOST, server.port)
+
+    def test_ciphersuites(self):
+        # Test unrecognized TLS 1.3 cipher suite name
+        with self.assertRaisesRegex(ssl.SSLError,
+                                    "No cipher suite can be selected"):
+            with socket.socket(socket.AF_INET) as sock:
+                s = test_wrap_socket(sock, cert_reqs=ssl.CERT_NONE,
+                                     ciphersuites="XXX",
+                                     min_version=ssl.TLSVersion.TLSv1_3)
```

Comment:
```suggestion
        with (
            socket.socket(socket.AF_INET) as sock,
            self.assertRaisesRegex(ssl.SSLError,
                                   "No cipher suite can be selected")
        ):
            s = test_wrap_socket(sock, cert_reqs=ssl.CERT_NONE,
                                 ciphersuites="XXX",
                                 min_version=ssl.TLSVersion.TLSv1_3)
```

That way, socket.socket() is not in the exception manager.